internal/definitions/definition_handler_contracts.go

```diff
@@ -30,96 +30,111 @@ func (dh *definitionHandlers) persistFFI(ctx context.Context, ffi *fftypes.FFI)
 		return false, nil
 	}
 
-	err = dh.database.UpsertFFI(ctx, ffi)
-	if err != nil {
-		return false, err
-	}
-
-	for _, method := range ffi.Methods {
-		err := dh.database.UpsertFFIMethod(ctx, method)
+	err = dh.database.RunAsGroup(ctx, func(ctx context.Context) error {
```

Comment:
I believe system broadcasts are already executed in RunAsGroup by the aggregator, so shouldn't need this here.

Reply:
You are right. It is way up the call stack but in `aggregator.go`,  all of this happens inside here:
```golang
func (ag *aggregator) processPinsDBGroup(items []fftypes.LocallySequenced) (repoll bool, err error) {
	pins := make([]*fftypes.Pin, len(items))
	for i, item := range items {
		pins[i] = item.(*fftypes.Pin)
	}
	err = ag.database.RunAsGroup(ag.ctx, func(ctx context.Context) (err error) {
		err = ag.processPins(ctx, pins)
		return err
	})
	return false, err
}
```